internal/message/subtrie.go

```diff
@@ -129,6 +143,30 @@ func (t *Trie) Lookup(ssid Ssid, filter func(s Subscriber) bool) (subs Subscribe
 	return
 }
 
+func (t *Trie) mqttLookup(query Ssid, subs *Subscribers, node *node, filter func(s Subscriber) bool) {
```

Comment:
We'll also need extensive test cases around this and make sure that this doesn't break the existing behavior.